include/deal.II/base/standard_tensors.h

```diff
@@ -0,0 +1,208 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__sstandard_tensors_h
+#define dealii__sstandard_tensors_h
+
+
+#include <deal.II/base/symmetric_tensor.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A collection of tensor definitions that conform to standard notation used
+ * in (nonlinear) elasticity. These hold specifically for the codimension 0
+ * case, where the metric tensor is the identity tensor.
+ *
+ * References for this notation include:
+ *    - G.A. Holzapfel (2001),
+ *      Nonlinear Solid Mechanics. A Continuum Approach for Engineering,
+ *      John Wiley & Sons.
+ *      ISBN: 0-471-82304-X
+ *    - P. Wriggers (2008)
+ *      Nonlinear finite element methods,
+ *      Springer Berlin Heidelberg.
+ *      ISBN: 978-3-540-71000-4
+ *      doi: 10.1007/978-3-540-71001-1
+ *
+ * @ingroup geomprimitives
+ * @relates SymmetricTensor
+ * @author Jean-Paul Pelteret, Andrew McBride, 2016
+*/
+template <int dim, typename Number>
+class StandardTensors
+{
+public:
+
+  /**
+   * The second-order referential/spatial symmetric identity tensor $\mathbf{I}$.
+   *
+   * This is defined such that, for any second-order SymmetricTensor,
+   * the following holds:
+   * $\mathbf{I} \cdot \{ \bullet \} = \{ \bullet \} \cdot \mathbf{I} =
+   * \{ \bullet \}$ and
+   * $\mathbf{I} : \{ \bullet \} = \textrm{trace} \{ \bullet \}$.
+   * This aligns with the second-order SymmetricTensor returned by unit_symmetric_tensor.
```

Comment:
add `()` to the function name at the end of the line so doxygen cross references it